mbed_lstools/lstools_base.py

```diff
@@ -61,10 +64,52 @@ def __init__(self, **kwargs):
         platform_dbs.append(LOCAL_PLATFORM_DATABASE)
         self.plat_db = PlatformDatabase(platform_dbs,
                                         primary_database=platform_dbs[0])
+        self.list_unmounted = list_unmounted
 
         if 'skip_retarget' not in kwargs or not kwargs['skip_retarget']:
             self.retarget()
 
+    @abstractmethod
+    def find_candidates(self):
+        """Find all candidate devices connected to this computer
+
+        Note: Should not open any files
+
+        @return A dict with the keys 'mount_point', 'serial_port' and 'target_id'
+        """
+        raise NotImplemented
+
+    def list_mbeds(self):
+        """ List details of connected devices
+        @return Returns list of structures with detailed info about each mbed
+        @details Function returns list of dictionaries with mbed attributes 'mount_point', TargetID name etc.
+        Function returns mbed list with platform names if possible
+        """
+        candidates = list(self.find_candidates())
+        logger.debug("Candidates for display %r", candidates)
+        result = []
+        for device in candidates:
+            if not device['mount_point'] :
+                if  (device['target_id'] and device['serial_port'] and
+                     not self.list_unmounted):
+                    logger.warning(
+                        "MBED with target id '%s' is connected, but not mounted. "
+                        "Use the '-u' flag to include it in the list.",
+                        device['target_id'])
+            else:
+                htm_target_id = self.read_htm_target_id(device['mount_point'])
+                if htm_target_id:
+                    logging.debug("Found more accurate target id %s -> %s",
+                                  device['target_id'], htm_target_id)
```

Comment:
I _think_ this log may be a bit misleading, as the target ID returned from the USB info can be the same as the target ID returned from the htm file.

I'd recommend only printing out this log only if they differ.

Reply:
This is a debug log. Users should not see this unless they ask for debug output.

Reply:
The fact also remains that we did find a more accurate id, which happens to be the same id.

Reply:
For debugging purposes, this log output *has been helpful*. Just seeing that it did this step is good.